cmd/openshift-tuned.go

```diff
@@ -652,128 +411,103 @@ func timedTunedReloader(tuned *tunedState) (err error) {
 	return err
 }
 
-func pullLabels(clientset *kubernetes.Clientset, tuned *tunedState, nodeName string) error {
-	// Resync period elapsed, force-pull node and pod labels
-	glog.V(2).Infof("Pulling node labels")
-	nodeLabels, err := nodeLabelsGet(clientset, nodeName)
-	if err != nil {
-		return err
-	}
-	if !reflect.DeepEqual(nodeLabels, tuned.nodeLabels) {
-		tuned.nodeLabels = nodeLabels
-		tuned.change.node = true
-	}
-
-	nowUnix := time.Now().Unix()
-	if nowUnix >= tuned.podLabelsPullTime {
-		glog.V(2).Infof("Pulling pod labels")
-		// Pull for pod labels was done >= resyncPeriodPodDefault
-		podLabels, err := podLabelsGet(clientset, nodeName)
-		if err != nil {
-			return err
-		}
-		tuned.setNextPodLabelsPullTime(nowUnix)
-		if !reflect.DeepEqual(podLabels, tuned.podLabels) {
-			tuned.podLabels = podLabels
-			tuned.change.pod = true
-		}
-	}
-
-	return nil
-}
-
-func pullResyncPeriod() int64 {
-	var (
-		err                  error
-		resyncPeriodDuration int64 = resyncPeriodNodeDefault
-	)
-	if os.Getenv("RESYNC_PERIOD") != "" {
-		resyncPeriodDuration, err = strconv.ParseInt(os.Getenv("RESYNC_PERIOD"), 10, 64)
-		if err != nil {
-			glog.Errorf("Error: cannot parse RESYNC_PERIOD (%s), using %d", os.Getenv("RESYNC_PERIOD"), resyncPeriodNodeDefault)
-			resyncPeriodDuration = resyncPeriodNodeDefault
-		}
+func getTunedProfile(obj interface{}) (profile *tunedv1.Profile, err error) {
+	profile, ok := obj.(*tunedv1.Profile)
+	if !ok {
+		return nil, fmt.Errorf("could not convert object to a tuned profile object: %+v", obj)
 	}
-
-	return resyncPeriodDuration
-}
-
-func pullResyncPeriodWithJitter() int64 {
-	resyncPeriodDuration := pullResyncPeriod()
-
-	// Add some randomness to the resync period, so that we don't end up in lockstep querying the API server
-	return resyncPeriodDuration + getJitter(resyncPeriodDuration, 0.3)
+	return profile, nil
 }
 
-func (tuned *tunedState) setNextPodLabelsPullTime(secsSinceEpoch int64) {
-	tuned.podLabelsPullTime = secsSinceEpoch + resyncPeriodPodDefault + getJitter(resyncPeriodPodDefault, 0.3) // try to avoid lockstep
-}
-
-func (resyncPeriod *resync) changeWatcher() (err error) {
+func changeWatcher() (err error) {
 	var (
 		tuned tunedState
-		wPod  watch.Interface
 		lStop bool
 	)
 
+	namespace := "openshift-cluster-node-tuning-operator"
 	nodeName := flag.Args()[0]
+	fieldSelector := fields.SelectorFromSet(fields.Set{"metadata.name": nodeName})
```

Comment:
You can toss the `nodeName` and `fieldSelector` in the `var` block as well for consistency